modules/sfp_fsecure_riddler.py

```diff
@@ -0,0 +1,207 @@
+            'Content-Type': 'application/json',
+        }
+
+
+        res = self.sf.fetchUrl('https://riddler.io/api/search',
+                               postData=json.dumps(params),
+                               headers=headers,
+                               useragent=self.opts['_useragent'],
+                               timeout=self.opts['_fetchtimeout'])
+
+        time.sleep(1)
+
+        if res['code'] in [ "400", "401", "402", "403" ]:
+            self.sf.error('Unexpected HTTP response code: ' + res['code'], False)
+            self.errorState = True
+            return None
+
+        if res['content'] is None:
+            return None
+
+        try:
+            data = json.loads(res['content'])
+        except BaseException as e:
+            self.sf.debug("Error processing JSON response from F-Secure Riddler: " + str(e))
+            return None
+
+        if not data:
+            self.sf.debug("No results found for " + qry)
+            return None
+
+        return data
+
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        if self.errorState:
+            return None
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        if srcModuleName == 'sfp_fsecure_riddler':
+            self.sf.debug("Ignoring " + eventData + ", from self.")
+            return None
+
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+
+        if self.opts['username'] == '' or self.opts['password'] == '':
+            self.sf.error('You enabled sfp_fsecure_riddler but did not set an API username/password!', False)
+            self.errorState = True
+            return None
+
+        if not self.token:
+            self.login()
+
+        self.results[eventData] = True
+
+        data = None
+
+        if eventName in ['INTERNET_NAME', 'DOMAIN_NAME']:
+            data = self.query("pld:" + eventData)
+        elif eventName == 'IP_ADDRESS':
+            data = self.query("ip:" + eventData)
+
+        if not data:
+            self.sf.info("No results found for " + eventData)
+            return None
+
+        e = SpiderFootEvent('RAW_RIR_DATA', str(data), self.__name__, event)
+        self.notifyListeners(e)
+
+        domains = list()
+        addrs = list()
+        coords = list()
+
+        for result in data:
+            host = result.get('host')
+
+            if not host:
+                continue
+
+            if not self.getTarget().matches(host, includeChildren=True, includeParents=True):
+                continue
+
+            domains.append(host)
+
+            addr = result.get('addr')
+
+            if addr:
+                addrs.append(addr)
+
+            coord = result.get('cordinates')
+
+            if coord:
+                coords.append(str(coord[0]) + ', ' + str(coord[1]))
```

Comment:
I would prefer we didn't trust/assume there will always be two elements and perform a quick sanity check.

Reply:
Fixed